client/settings/settings.jsx

```diff
@@ -88,16 +112,52 @@ export default class Settings extends React.Component {
           <AppSettings
             localSettings={local}
             formRef={this._form}
+            onChange={this.onSettingsChange}
             onSubmit={this.onSettingsSubmit}
           />
         )
-      case TAB_GAME:
+      case isRemastered ? SCR_TAB_INPUT : V1161_TAB_INPUT:
```

Comment:
I honestly did not know you could do this kind of thing in a case statement and I am mildly scared :P

Reply:
Might be better to move away from this idea that each of our settings tabs would support both remastered and 1.16, and instead write dedicated components for each (then you'd just have separate case statements here instead of this complexity)

Reply:
Thinking some more, is it even really necessary to have separate identifiers for these things? We only display one or the other anyway

Reply:
Yeah, this all started by trying to hide Sound and Gameplay tabs from the 1.16.1 version since we don't have any of those settings for that version yet. So the the 1.16.1 version only had 3 tabs (App, Input, and Video) and the SC:R had all 5 (App, Input, Sound, Video, and Gameplay).

The first issue I noticed was due to the way I was rendering tab items, which was (I had to use the fragment instead of `null` when not rendering anything, since tab component is cloning the children to attach some props to them):
```js
<Tabs activeTab={activeTab} onChange={this.onTabChange}>
  <TabItem key='app' text='App' />
  <TabItem key='input' text='Input' disabled={isTabDisabled} />
  {isRemastered ? <TabItem key='sound' text='Sound' disabled={isTabDisabled} /> : <></>}
  <TabItem key='video' text='Video' disabled={isTabDisabled} />
  {isRemastered ? <TabItem key='gameplay' text='Gameplay' disabled={isTabDisabled} /> : <></>}
</Tabs>
```
The problem with this approach was that the tab component would incorrectly calculate the tab indicator size, since it calculates that based on the children count (which counts fragments as well). So the indicator size would be wrong in the 1.16.1 version since it only displayed 3 tabs, but counted 5.

So the next thing I did was I dynamically created the array of tabs based on the version and that fixed the indicator size problem. However, now the switch/case statement which used the tab value (which is just an index in an array), would incorrectly display the contents since the 1.16.1 version had fewer tabs. E.g. Video is the 4th tab in the SC:R version and 3rd tab in the 1.16.1 version. That's what cause the switch/case statement above to become weirdly complicated.

The _proper_ fix for this would be upgrading the Tab (and TabItem) components and make each TabItem have a defined value, so that we don't rely on array indices as its implicit value which we use to detect an active item here:
https://github.com/ShieldBattery/ShieldBattery/blob/master/client/material/tabs.jsx#L122

At the time I wrote the tab component I thought the usage of tabs will remain simple enough to not warrant doing that (as we already do in the Select component afaik). And I really didn't feel like doing that in this PR because this "bug" was the last thing I was fixing in it and I was already too tired at this point :d

If you think we should do it anyway in this PR, then I can muster up some mental fortitude and get it over with :p

Reply:
Not sure I quite understand how this affects the switch statement, but I assume you know better than I do :) I'm fine with this for this PR, would probably just file an issue to make that change to the tabs component later